frame/ethereum/src/lib.rs

```diff
@@ -393,16 +394,17 @@ pub mod pallet {
 }
 
 impl<T: Config> Pallet<T> {
-	pub fn transaction_weight(transaction_data: &TransactionData) -> (Option<Weight>, Option<u64>) {
-		match <T as pallet_evm::Config>::GasWeightMapping::gas_to_weight(
+	pub fn transaction_pov(transaction_data: &TransactionData) -> (Option<Weight>, u64){
+		let weight_limit = <T as pallet_evm::Config>::GasWeightMapping::gas_to_weight(
 			transaction_data.gas_limit.unique_saturated_into(),
 			true,
-		) {
-			weight_limit if weight_limit.proof_size() > 0 => (
-				Some(weight_limit),
-				Some(transaction_data.proof_size_base_cost()),
-			),
-			_ => (None, None),
+		);
+		let proof_size_pre_execution = get_proof_size().unwrap_or_default();
+
+		if weight_limit.proof_size() > 0 {
+			(Some(weight_limit), proof_size_pre_execution)
+		} else {
+			(None, 0)
```

Comment:
Should be `(None, None)`
```suggestion
			(None, None)
```

Otherwise it breaks the assumptions of `validate_common` in `CheckEvmTransaction`